x-pack/platform/plugins/shared/index_management/public/application/sections/home/index_list/details_page/details_page_overview/details_page_overview.tsx

```diff
@@ -104,20 +109,30 @@ export const DetailsPageOverview: React.FunctionComponent<Props> = ({ indexDetai
     });
   }, [cloud]);
 
+  useEffect(() => {
+    const subscription = licensing?.license$.subscribe((license: ILicense) => {
+      setIsEnterpriseLicense(license.isActive && license.hasAtLeast('enterprise'));
+    });
```

Comment:
the index_management plugin already has a license subscription [see here](https://github.com/elastic/kibana/blob/main/x-pack/platform/plugins/shared/index_management/public/plugin.ts#L249-L252) you could add a `canUseEIS` value and just read that from AppContext like [canUseSyntheticSource](https://github.com/elastic/kibana/blob/main/x-pack/platform/plugins/shared/index_management/public/application/components/mappings_editor/mappings_editor.tsx#L63-L64)